lib/utils.js

```diff
@@ -167,6 +167,20 @@ async function clearSystemFiles (wda) {
   }
   derivedDataCleanupMarkers.set(logsRoot, 0);
 
+  // Cleaning up big temporary files created by XCTest: https://github.com/appium/appium/issues/9410
+  const cleanupTask = new SubProcess('bash', ['-c',
+    `find -E /private/var/folders ` +
+    `-regex '.*/Session-WebDriverAgentRunner.*\\.log$|.*/StandardOutputAndStandardError\\.txt$' ` +
+    `-type f -exec sh -c 'echo "" > "{}"' \\;`], {
+      detached: true,
+      stdio: ['ignore', 'pipe', 'pipe'],
+    }
+  );
+  await cleanupTask.start(0);
+  // Do not wait for the task to be completed, since it might take a lot of time
+  // We keep it running after Appium process is killed
+  cleanupTask.proc.unref();
+
```

Comment:
At the very least there ought to be some error reporting in here. It might be nice to report what was actually done, too, if that is possible.

Reply:
added logging